### PR TITLE
Add receiver null check for linkToSpecial case

### PR DESCRIPTION
**Pull request by fengxue-IS**
Fixes: #12571 

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

**Comment by fengxue-IS** [comment]
This PR fixes the `testBindTo` testcase.
As per comments in #12571, tests that check for `isJitted` is checking for internal behavior and should be excluded.

@tajila can you please take a look

File: runtime/vm/BytecodeInterpreter.hpp [hunk 1]
```diff
@@ -8355,10 +8355,21 @@ class INTERPRETER_CLASS
...
+		if (J9_ARE_NO_BITS_SET(romMethod->modifiers, J9AccStatic)) {
+			j9object_t mhReceiver = ((j9object_t *)_sp)[methodArgCount - 1];
+			if (J9_UNEXPECTED(NULL == mhReceiver)) {
+				if (fromJIT) {
+					/* Restore SP to before popping memberNameObject. */
+					_sp -= 1;
+					buildJITResolveFrame(REGISTER_ARGS);
+				}
+				return THROW_NPE;
+			}
```
Comment (DanHeidinga) [inline runtime/vm/BytecodeInterpreter.hpp:8369]: This section duplicates the `THROW_NPE` section above.  The two can be commoned by adding a `throw_npe:` to the first version of this code and replacing this with `goto throw_npe;`

**Comment by DanHeidinga** [comment]
jenkins test sanity winojdk292 jdk16 depends ibmruntimes/openj9-openjdk-jdk16#29

**Comment by fengxue-IS** [comment]
@DanHeidinga I've fixed the compile error with undefined name after moving the variables to avoid goto cross initialization.
Can you please restart the build.

**Comment by DanHeidinga** [comment]
> I've fixed the compile error with undefined name after moving the variables to avoid goto cross initialization.

That's why I suggested keeping the original location and jumping back to it as it avoids the initialization issue.

ie:
```c
if (J9_UNEXPECTED(NULL == mhReceiver)) {
_throw_npe:
	if (fromJIT) {
		/* Restore SP to before popping memberNameObject. */
		_sp -= 1;
		buildJITResolveFrame(REGISTER_ARGS);
	}
	return THROW_NPE;
}

....

if (null) {
   goto _throw_npe;
}
```
You'll see the same pattern in the BytecodeInterpreter as well for the same reason

**Comment by DanHeidinga** [comment]
jenkins test sanity winojdk292 jdk16 depends ibmruntimes/openj9-openjdk-jdk16#29

**Comment by fengxue-IS** [comment]
test failure in https://openj9-jenkins.osuosl.org/job/Test_openjdk16_j9_sanity.functional_x86-64_windows_ojdk292_Personal/10/ is due to conflict with #12668, the check is trying to check `vm->initialMethods.throwDefaultConflict`'s romMethod which is causing the crash.

I have updated the code to handle the default conflict case correctly

**Comment by DanHeidinga** [comment]
jenkins test sanity winojdk292 jdk16 depends ibmruntimes/openj9-openjdk-jdk16#29

**Review by tajila — APPROVED** [review]

**Review by DanHeidinga — APPROVED** [review]

**Pull request merged**